Exercises/chapter-11/e3-promise-all.js

```diff
@@ -0,0 +1,47 @@
+function Promise_all(promises) {
+    return new Promise((resolve, reject) => {
+        const results = [];
+        try {
+            if (promises.length === 0) {
```

Comment:
No need to special case this. It won't speed the code up or make it easier to read.